### PR TITLE
Added implementation for routing layer

### PR DESCRIPTION
**Pull request by catchashu10**

### CODE REVIEW

File: common/router.cc [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    if (topic_partitions_.count(topic)) {
+        return topic_partitions_[topic];
+    }
+    throw std::runtime_error("Partition count not available for topic: " + topic);
+}
+
+// Fetch metadata for a given topic only if it is not already cached
+// If using cached metadat then need to ensure that cache is updated periodically
+// Another option is to always fetch metadata from the server insted of checking if cache exists.
+void Router::UpdateMetadata(const std::string& topic) {
```
Comment (catchashu10) [inline common/router.cc:52]: @Pratik-Sanghavi @yadavpa1 What do you think about this logic? We can use cached info in `routing_table_` to check if we should fetch metadata for the topic from server of not. But we need to find a way to refresh cache or update cache if the leader address present is not a valid one.

Comment (catchashu10) [reply common/router.cc:52]: Commented out the cache checking logic. For now every time producer will fetch the meta data from server before sending produce request to the server.

Comment (Pratik-Sanghavi) [reply common/router.cc:52]: LGTM

**Review by Pratik-Sanghavi — APPROVED** [review]
Looks good to me

**Pull request merged**